APNGImageSerialization/Classes/APNGImageSerialization.m

```diff
@@ -266,35 +305,88 @@ + (UIImage *)animatedImageNamed:(NSString *)name
     }
     if (path) {
         NSData *data = [NSData dataWithContentsOfFile:path];
-        return UIAnimatedImageWithAPNGData(data, 0.f, scale, NULL);
+        if (AnimatedPngDataIsValid(data)) {
+            return UIAnimatedImageWithAPNGData(data, scale, 0.f, nil);
+        }
     }
-    return nil;
+    return [self apng_animatedImageNamed:name];
 }
 
-+ (UIImage *)apng_animatedImageWithAPNGData:(NSData *)data
+
++ (UIImage *)apng_imageWithContentsOfFile:(NSString *)path __attribute__((objc_method_family(new)))
 {
-    return UIAnimatedImageWithAPNGData(data);
+    if (path) {
+        NSData *data = [NSData dataWithContentsOfFile:path];
+        if (AnimatedPngDataIsValid(data)) {
+            if ([[path stringByDeletingPathExtension] hasSuffix:@"@2x"]) {
```

Comment:
here should be 3x.
In fact, I'm not swizzling those method because APNG is rarely used, Developers should call specific method explicitly to initialize a APNG instance. Swizzling can make a performance loss, most of the case it's just a normal PNG file

Reply:
That’s fair that it’s an obscure file type. We’re attempting to replace gifs on our platform. This lets us use it as a drop in replacement on top of AlamofireImage

In terms of performance cost, it should be trivially slower since there’s a tiny number of additional function calls on top of the much larger cost of deserializing data. That *is* why I didn’t replace `+imageNamed` since that would invalidate UIKit’s image cache.

That being said, feel free to close this if you don’t feel like it’s the right direction for this library.